[quant][graphmode] Fold quantized prepacking ops

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #35571 [quant][graphmode] Quantization support for at::sort
* #35558 [quant][graphmode][refactor] getGeneralTensorInputs(Node*) -> getPassThroughInputs(Value*)
* #35557 [quant][graphmode][fix] docs for InsertObservers
* #35509 [quant][graphmode] Quantization support for `quantized::add_scalar_relu` and `quantized::add_scalar_relu_out`
* #35385 [quant][graphmode] Remove unused patterns
* #35347 [quant][graphmode] Quantization support for `aten::conv3d`
* #35345 [quant][graphmode][fix] Add filter for `quantized::add`
* #35336 [quant][graphmode] qconfig_dict support None
* #35420 [quant][graphmode] Make `aten::relu` a general op
* #35335 [quant][graphmode] `prepare_script` takes original qconfig_dict
* #35334 [quant][graphmode] Quantization support for `quantized::add_scalar`
* #35333 [quant][graphmode][refactor] Support filter function in quant fusion patterns
* #35332 [quant][graphmode] Quantization support for permute and repeat_interleave
* #35130 [quant][graphmode] Make interpolate/upsample work again
* #35142 [quant][graphmode] SwapDeQuant support prim::If
* #35141 [quant][graphmode][refactor] getGeneralOpTensorInputIndexes -> getGeneralOpTensorInputs
* #35135 [quant][graphmode][refactor] swapDeQuant takes block as arugment
* **#35077 [quant][graphmode] Fold quantized prepacking ops**

Summary:
Fold the prepack ops: `quantized::linear_prepack` and `quantized::conv2d_prepack` after
`freeze`

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20655301](https://our.internmc.facebook.com/intern/diff/D20655301)

jerryzh168 committed:
[quant][graphmode] Fold quantized prepacking ops

Summary:
Fold the prepack ops: `quantized::linear_prepack` and `quantized::conv2d_prepack` after
`freeze`

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Fold quantized prepacking ops"


Summary:
Fold the prepack ops: `quantized::linear_prepack` and `quantized::conv2d_prepack` after
`freeze`

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Fold quantized prepacking ops"


Summary:
Fold the prepack ops: `quantized::linear_prepack` and `quantized::conv2d_prepack` after
`freeze`

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

kimishpatel commented on torch/csrc/jit/passes/quantization.cpp:
```diff
@@ -2560,6 +2561,15 @@ void DedupModuleUses(Module& module) {
   d.dedup();
 }
 
+void FoldQuantizedPrepackingOps(Module& module) {
```
I thought coding convention was camelcase for function names.

kimishpatel replied:
I guess I also have mixed that up on some of my passes.

jerryzh168 replied:
yeah we are following camelcase for methods and helper functions I think. For jit passes I saw we are using capital case, e.g.: https://bddppq.github.io/codebrowser/pytorch/pytorch/torch/csrc/jit/passes/constant_propagation.h.html

jerryzh168 committed:
Update on "[quant][graphmode] Fold quantized prepacking ops"


Summary:
Fold the prepack ops: `quantized::linear_prepack` and `quantized::conv2d_prepack` after
`freeze`

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

kimishpatel reviewed (approved):
LGTM.

jerryzh168 committed:
Update on "[quant][graphmode] Fold quantized prepacking ops"


Summary:
Fold the prepack ops: `quantized::linear_prepack` and `quantized::conv2d_prepack` after
`freeze`

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Fold quantized prepacking ops"


Summary:
Fold the prepack ops: `quantized::linear_prepack` and `quantized::conv2d_prepack` after
`freeze`

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Fold quantized prepacking ops"


Summary:
Fold the prepack ops: `quantized::linear_prepack` and `quantized::conv2d_prepack` after
`freeze`

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Fold quantized prepacking ops"


Summary:
Fold the prepack ops: `quantized::linear_prepack` and `quantized::conv2d_prepack` after
`freeze`

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Fold quantized prepacking ops"


Summary:
Fold the prepack ops: `quantized::linear_prepack` and `quantized::conv2d_prepack` after
`freeze`

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Fold quantized prepacking ops"


Summary:
Fold the prepack ops: `quantized::linear_prepack` and `quantized::conv2d_prepack` after
`freeze`

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Fold quantized prepacking ops"


Summary:
Fold the prepack ops: `quantized::linear_prepack` and `quantized::conv2d_prepack` after
`freeze`

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Fold quantized prepacking ops"


Summary:
Fold the prepack ops: `quantized::linear_prepack` and `quantized::conv2d_prepack` after
`freeze`

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Fold quantized prepacking ops"


Summary:
Fold the prepack ops: `quantized::linear_prepack` and `quantized::conv2d_prepack` after
`freeze`

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Fold quantized prepacking ops"


Summary:
Fold the prepack ops: `quantized::linear_prepack` and `quantized::conv2d_prepack` after
`freeze`

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Fold quantized prepacking ops"


Summary:
Fold the prepack ops: `quantized::linear_prepack` and `quantized::conv2d_prepack` after
`freeze`

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Fold quantized prepacking ops"


Summary:
Fold the prepack ops: `quantized::linear_prepack` and `quantized::conv2d_prepack` after
`freeze`

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Fold quantized prepacking ops"


Summary:
Fold the prepack ops: `quantized::linear_prepack` and `quantized::conv2d_prepack` after
`freeze`

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Fold quantized prepacking ops"


Summary:
Fold the prepack ops: `quantized::linear_prepack` and `quantized::conv2d_prepack` after
`freeze`

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Fold quantized prepacking ops"


Summary:
Fold the prepack ops: `quantized::linear_prepack` and `quantized::conv2d_prepack` after
`freeze`

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20655301](https://our.internmc.facebook.com/intern/diff/D20655301)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Fold quantized prepacking ops"


Summary:
Fold the prepack ops: `quantized::linear_prepack` and `quantized::conv2d_prepack` after
`freeze`

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20655301](https://our.internmc.facebook.com/intern/diff/D20655301)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Fold quantized prepacking ops"


Summary:
Fold the prepack ops: `quantized::linear_prepack` and `quantized::conv2d_prepack` after
`freeze`

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20655301](https://our.internmc.facebook.com/intern/diff/D20655301)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Fold quantized prepacking ops"


Summary:
Fold the prepack ops: `quantized::linear_prepack` and `quantized::conv2d_prepack` after
`freeze`

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20655301](https://our.internmc.facebook.com/intern/diff/D20655301)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Fold quantized prepacking ops"


Summary:
Fold the prepack ops: `quantized::linear_prepack` and `quantized::conv2d_prepack` after
`freeze`

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20655301](https://our.internmc.facebook.com/intern/diff/D20655301)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Fold quantized prepacking ops"


Summary:
Fold the prepack ops: `quantized::linear_prepack` and `quantized::conv2d_prepack` after
`freeze`

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20655301](https://our.internmc.facebook.com/intern/diff/D20655301)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Fold quantized prepacking ops"


Summary:
Fold the prepack ops: `quantized::linear_prepack` and `quantized::conv2d_prepack` after
`freeze`

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20655301](https://our.internmc.facebook.com/intern/diff/D20655301)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Fold quantized prepacking ops"


Summary:
Fold the prepack ops: `quantized::linear_prepack` and `quantized::conv2d_prepack` after
`freeze`

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20655301](https://our.internmc.facebook.com/intern/diff/D20655301)

[ghstack-poisoned]
